src/lib/mod.rs

```diff
@@ -43,29 +43,29 @@ pub type AnyhowResult<T = ()> = anyhow::Result<T>;
 pub enum TargetCanister {
     Governance(Principal),
     Ledger(Principal),
+    Swap(Principal),
+    IcpLedger,
 }
 
 impl From<TargetCanister> for Principal {
     fn from(target_canister: TargetCanister) -> Self {
         match target_canister {
             TargetCanister::Governance(principal) => principal,
             TargetCanister::Ledger(principal) => principal,
+            TargetCanister::Swap(principal) => principal,
+            TargetCanister::IcpLedger => "ryjl3-tyaaa-aaaaa-aaaba-cai".parse().unwrap(),
```

Comment:
Instead of using String could you use the const `CanisterId` for ledger and convert it to a `Principal`? You can find it here in [nns/constants](https://gitlab.com/dfinity-lab/public/ic/-/blob/master/rs/nns/constants/src/lib.rs#L60-61)